server/src/main/java/io/crate/execution/dml/TranslogIndexer.java

```diff
@@ -138,6 +145,12 @@ private IndexDocumentBuilder populateLuceneFields(BytesReference source) throws
             entry.setValue(castValue);
```

Comment:
This bit here also only works for top-level entries in the source map, which is the same bug as @seut found in the first place for index entries.

Reply:
I think for object it will sanitize the whole object, as `ObjectType.sanitizeValueLenient` goes through all sub-columns, so should be fine?

Not yet sure about the final approach but afaik any solution should use sanitized value, we can't access docMap as is,so I was going to keep this bit

Reply:
Yes, sorry, I was incorrect here.